### ISSUE TITLE
Activate me on Classic SharePoint 201x page! TypeError: Object doesn't support property or method 'debug'

### ISSUE THREAD

**Issue by f-elder** [365CSI/iCSR#16]
I receive this error when attempting to activate on a SharePoint online page in Classic mode.  Any ideas on a resolution?

**Comment by Danny-Engelman** [comment]
Sounds like the old IE error.
Where a line 
```
console.debug( ... )
```
is left in the JS code

There is no such line in this repo, so you have to dig into other sources yourself

**Comment by f-elder** [comment]
Receive "ReferenceError: g_365CSImodule is not defined" in Chrome on the same page.

**Comment by Danny-Engelman** [comment]
``g_365CSImodule`` is only in JS code you pull directly from my personal server.

Check the Network tab; what files are you pulling in from my 365CSI.nl domain??

**Suggested solution:**

* Download the JS library from https://github.com/365CSI/iCSR
* save it to your own server
* use that file

Note: alas Microsoft has discontinued CSR/JSLink for SharePoint Online Modern Experiences; so you might want to reconsider if you want to continue with 'old' libraries.

**Comment by f-elder** [comment]
Yes, I downloaded the library and am trying to use it. Digging deeper I'm receiving "sharepoint mime type ('text/html') is not executable and strict mime type checking is enabled" for those files.  I've made sure they are all published, but still receive the error.  I'm working in classic mode, which Microsoft recently announced would continue to be supported.  I'm awaiting until SPFX matures to make another switch.

**Comment by Danny-Engelman** [comment]
You are **not** using the code from this iCSR repo.
https://github.com/365CSI/iCSR/blob/master/icsr.js
does **not** contain any ``g_365CSI`` related code

We first have to make sure you are using the correct JS library

Check your Browser Tools Network; which iCSR references are in there? (full paths)

**Comment by f-elder** [comment]
I am using the icsr.min.js and the associated minified files in the site.  The "sharepoint mime type ('text/html') is not executable and strict mime type checking is enabled" error precluded the files from being executed.  Consequently it went to the web.  I moved the files to another library to address that error, but am receiving an "Uncaught ReferenceError: iCSR is not defined" error using a sample.

The code is as follows:
iCSR.Me({
file :'~sitecollection/script/icsr/icsr.min.js',
"Due":{//date column with the task due date
 View : iCSR.DueDate({
  ranges:     "coral/red,-1,pink,0,khaki/red,1,lightgreen,2,yellowgreen",
  colortag:   "TR",
  red:        "style=color:[msRed]",
  labelNodate:"<i [red]>No Due Date!</i>",
  labelToday: "<b [red]>Today</b>",
})}});

**Comment by Danny-Engelman** [comment]
You now mention **yet another** problem, 

* debug not defined
* g_365CSI module not defined
* iCSR not defined

They are unrelated; and the first 2 can **not** occur on iCSR code from this repo.

The last one means the iCSR library is not loaded at all

So make sure you properly load that JS file, either as JSLink or as UserCustomAction

If you can not do that then I have to be harsh and say this library is over your head.

Note: alas Microsoft has discontinued CSR/JSLink for SharePoint Online Modern Experiences; so you might want to reconsider if you want to continue with 'old' libraries.

**Comment by f-elder** [comment]
As noted I'm running in classic mode and will until SPFX matures. In the interim I've explained the why it appears the referenced icsr.min.js file was not being loaded. It's unusual as I've not run into these issues on prior updates.  Using the custom user action the icsr.min.js file is now clearly referenced, but there is still an issue.

I used the SPUserCustomActionManagerOnline tool to load the custom user action for the icsr.min.js file without error.  However on the page the following errors are noted: "Uncaught ReferenceError: Type is not defined at icsr.min.js:1" and "Uncaught ReferenceError: iCSR is not defined at statusbydateicsr.js:1".  The code for the latter is as follows:

iCSR.Me({
"Due":{//date column with the task due date
 View : iCSR.DueDate({
  ranges:     "coral/red,-1,pink,0,khaki/red,1,lightgreen,2,yellowgreen",
  colortag:   "TR",
  red:        "style=color:[msRed]",
  labelNodate:"<i [red]>No Due Date!</i>",
  labelToday: "<b [red]>Today</b>",
})}});

The icsr.min.js file is in the Site Assets library and is referenced as follows:
"~sitecollection/SiteAssets/iCSR/icsr.min.js".

**Comment by Danny-Engelman** [comment]
the error states that iCSR is not defined... 

Check the F12 DevTools Network tab; is the JS filet referenced?

**Comment by f-elder** [comment]
I updated the reference to the ~site scope and they are resolving with 200s in the network view of the developer tools, but nothing changes in the row rendering. It's in the site collection Site Assets library, but that doesn't appear to be resolved properly with ~sitecollection.  Due is the visible and internal name of the date field used and it is of SPFieldDateTime type.  I see no changes under Styles or Computed. Under the Event Listeners I see it invoked in keydown, keyup, mousedown, mouseup, and resize.
